File: 2023/rust/src/main.rs
#![allow(dead_code)]
#![allow(unused_imports)]
mod days;
mod utils;

use std::time::Instant;

use days::day1;
use days::day10;
use days::day11;
use days::day12;
use days::day13;
use days::day14;
use days::day15;
use days::day16;
use days::day6;
use days::day8;
use days::day9;

fn main() {
    run_part1();
    run_part2();
}

fn run_part1() {
    let start = Instant::now();

    day16::part1();

    let elapsed = start.elapsed();

    println!("Time elapsed: {:?}\n", elapsed);
}

fn run_part2() {
    let start = Instant::now();

    day16::part2();

    let elapsed = start.elapsed();

    println!("Time elapsed: {:?}\n", elapsed);
}

File: 2023/typescript/main.ts
async function main() {
	await runPart1(20);
	await runPart2(20);
}

async function runPart1(day: number) {
	await import(`./Day ${day}/part1.ts`).then(async (part1) => {
		const start = performance.now();

		const result = await part1.solve();

		const end = performance.now();

		console.log(`\nSolution for Day ${day} Part 1: ${result}`);
		console.log(`Time elapsed: ${Math.round((end - start + Number.EPSILON) * 100) / 100}ms\n`);
	});
}

async function runPart2(day: number) {
	await import(`./Day ${day}/part2.ts`).then(async (part2) => {
		const start = performance.now();

		const result = await part2.solve();

		const end = performance.now();

		console.log(`Solution for Day ${day} Part 2: ${result}`);
		console.log(`Time elapsed: ${Math.round((end - start + Number.EPSILON) * 100) / 100}ms\n`);
	});
}

main();
